pkg/scheduler/internal/queue/scheduling_queue.go

```diff
@@ -446,6 +518,40 @@ func (p *PriorityQueue) flushBackoffQCompleted() {
 func (p *PriorityQueue) Pop() (*v1.Pod, error) {
 	p.lock.Lock()
 	defer p.lock.Unlock()
+	if util.EquivalenceClassEnabled() {
+		for {
```

Comment:
Add a summary comment here why this loop will not cause dead lock. You need to make sure other contributors will not break your assumptions by simply reading your comment.